tools/keepalived/keepalived/vrrp/vrrp_iproute.c

```diff
@@ -76,9 +112,20 @@ netlink_route(ip_route_t *iproute, int cmd)
             cmd, tmp_dst, iproute->dmask, tmp_src, iproute->ifname, iproute->scope);
     FREE(tmp_dst);
     FREE(tmp_src);
-    route_conf = (struct dp_vs_route_conf *)malloc(sizeof(struct dp_vs_route_conf));
-    dpvs_fill_rtconf(iproute, route_conf);
-    ipvs_set_route(route_conf, cmd);
+
+    if (iproute->dst->ifa.ifa_family == AF_INET) {
+        struct dp_vs_route_conf *route_conf;
+        route_conf = (struct dp_vs_route_conf *)malloc(sizeof(struct dp_vs_route_conf));
+        memset(route_conf, 0, sizeof(*route_conf));
+        dpvs_fill_rt4conf(iproute, route_conf);
+        ipvs_set_route(route_conf, cmd);
```

Comment:
I think it's better to free the memory 'route_conf/route6_conf' where it 's allocated.

Reply:
fixed